### PR TITLE
Skip UTF-8 BOM in FileStorage

### LINKED ISSUE #5073: Crash when trying to load utf-8 xml file with a BOM
Transferred from http://code.opencv.org/issues/4486

```
|| Lior Da on 2015-07-13 21:18
|| Priority: Normal
|| Affected: branch 'master' (3.0-dev)
|| Category: None
|| Tracker: Bug
|| Difficulty: 
|| PR: 
|| Platform: x64 / Windows
```
## Crash when trying to load utf-8 xml file with a BOM

```
When calling the next line:
cv::FileStorage root("_root.xml", cv::FileStorage::READ);
the program crashes.

If I change _root.xml to be encoded without BOM, it does not crash.
```

### PR DESCRIPTION
**Pull request by sovrasov**
Resolves #5073

### CODE REVIEW

File: modules/core/src/persistence.cpp [hunk 1]
```diff
@@ -1787,6 +1796,11 @@ icvYMLParse( CvFileStorage* fs )
 {
     char* ptr = fs->buffer_start;
     int is_first = 1;
+    ptr = icvYMLSkipSpaces( fs, ptr, 0, INT_MAX );
+    if(fs->buffer_end - ptr > 3)
```
Comment (alalek) [inline modules/core/src/persistence.cpp]: `>= 3` (in case of "empty" document)?

**Comment by vpisarev** [comment]
if I understand the fix correctly, you check for BOM regularly inside the file. Isn't it more correct to check it once in very beginning of the file and _before_ whitespaces (if any)?

**Comment by sovrasov** [comment]
I tried to do so, but in this case BOM check doesn't works at all because `SkipSpaces` introduces side effects even if no whitespaces found.

**Comment by sovrasov** [comment]
@vpisarev I implemented BOM skipping via changing of the `fs` object state. Hope it won't introduce hidden bugs.

**Comment by vpisarev** [comment]
thanks! now it looks good! :+1:

**Pull request merged**